blog/2020/05/rr.md

```diff
@@ -0,0 +1,452 @@
+      document.getElementById('player_main').addEventListener('animationend', () => {
+        document.getElementById('record_player').classList.add("hidden");
+        document.getElementById('replay_player').classList.remove("hidden");
+        document.getElementById('replay_player').play();
+      });
+    });
+  </script>
+~~~
+
+The Julia project, like any large open source project, gets a large number of
+bug reports everyday. As the developers of the language, we try our best to
+be as responsive as possible and to triage, investigate and fix any bugs as
+quickly as possible. For some bugs, this is easy. If the bug report is well
+written and the problem is evident, getting it fixed is usually a quick affair.
+However, for a large number of issues, this is not as easy. There are several
+common reasons why bugs may go unaddressed for extended periods of time, for
+example:
+
+1) The bug may not reproduce deterministically, or maybe only reproduces on the
+   reporter's machine (sometimes known as a [Heisenbug](https://en.wikipedia.org/wiki/Heisenbug)).
+2) The bug report may be incomplete as to the environment in which the bug occurs,
+   making it hard to reproduce.
+3) The bug reporter may have seen the bug once, but not been entirely sure what
+   caused it making it hard to give a reproducer and making the bug report
+   largely unactionable.
+4) The bug may occur only in a large project that is hard to set up.
+5) The bug may require specialized expertise to diagnose (e.g. a missing GC root),
+   but the time commitment to reproduce it is too large for those who have the
+   requisite expertise, since their time is in high demand.
+
+In addition, there are the bugs that never get filed, because the user may
+feel the effort to write a high-quality bug report is too high, since their
+specific setup would be too complicated to describe and reproduce. Such
+experiences are frustrating both for the users encountering them and for us,
+because we don't learn about them and sometimes only hear about it years later,
+when we get told that somebody gave up on their project in Julia because they
+encountered some crashing problem that they didn't feel qualified to reduce into
+a concise bug report and subsequently gave up on using Julia for their project.
+Lastly, we don't want or expect our users to be expert bug reporters. They are
+often working scientists who are good programmers, but may not have any
+background in software engineering. These are some of our most valuable users,
+and we want to make sure their bugs are addressed.
+
+For users who did encounter such particularly difficult problems, we have for
+a long time had one particular answer: If you can reproduce it on a linux
+machine and get us a trace from the `rr` tool <https://rr-project.org/>, we can
+probably get it fixed for you very rapidly. For the uninitiated, `rr` is a Linux
+debugging tool originally written by Robert O'Callahan at Mozilla. It is a tool
```

Comment:
This is a bit misleading as written, I didn't write much actual rr code before it became a usable tool :-). Maybe write "originally developed at Mozilla by Robert O'Callahan and others"?

Reply:
Thanks, I wasn't quite aware of the history, applied as suggested.